scripts/flux-build.sh

```
@@ -99,10 +101,31 @@ for ks_file in $(find "$CLUSTER_DIR" -maxdepth 3 -name '*.yaml' \
+        export "$key"="$val"
+        LITERAL_KEYS+=("$key")
+      done
+    fi
+
+    if RENDERED="$(kustomize build "$BUILD_PATH" "${KUSTOMIZE_FLAGS[@]}" | python3 -c "$ENVSUBST_PY")"; then
+      # kustomize build succeeding says nothing about substitution — check
+      # separately for ${VAR} placeholders that never got resolved.
+      UNRESOLVED="$(printf '%s' "$RENDERED" | grep -oE '\$\{[A-Za-z_][A-Za-z0-9_]*(:[^}]*)?\}' | sort -u || true)"
+      if [[ -n "$UNRESOLVED" ]]; then
+        echo "  ✗ $name — unresolved variable(s):"
+        echo "$UNRESOLVED" | sed 's/^/      /'
```

Shellcheck (0.11.0)

[style] See if you can use ${variable//search/replace} instead.

(SC2001)